From d63236bbeab8b5db7a013728a08d3a9d916c5035 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:28 -0400
Subject: [tools] perf/header: Allow header->data_offset to be predetermined

Message-id: <1408953210-26343-141-git-send-email-jolsa@redhat.com>
Patchwork-id: 88910
O-Subject: [PATCH RHEL7.1 BZ1133083 140/322] perf header: Allow header->data_offset to be predetermined
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit d645c442e68d24e64c46845bc8bb5d5a0a70b249
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Wed Dec 11 14:36:28 2013 +0200

description
===========
It will be necessary to predetermine header->data_offset to allow space
for attributes that are added later.  Consequently, do not change
header->data_offset if it is non-zero.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/header.c b/tools/perf/util/header.c
index 0bb830f..61c5421 100644
--- a/tools/perf/util/header.c
+++ b/tools/perf/util/header.c
@@ -2327,7 +2327,8 @@ int perf_session__write_header(struct perf_session *session,
   }
  }
 
- header->data_offset = lseek(fd, 0, SEEK_CUR);
+ if (!header->data_offset)
+  header->data_offset = lseek(fd, 0, SEEK_CUR);
  header->feat_offset = header->data_offset + header->data_size;
 
  if (at_exit) {
-- 
1.7.1